### DISCUSSION TITLE
Print Quality Posts

### DISCUSSION THREAD

**Discussion by CAP1Sup** [CAP1Sup/Intellistep#116] [Show and tell]
This is an area where people can post pictures of their prints. The prints are helpful to understand how the code is performing out in the wild and is a big help. Keep in mind to include:
- Photo with good detail (so that the layer lines are visible and it's easy to see any noticed imperfections in the part)
- Version of Intellistep (displayed on bootup)
- General settings (such as speed, acceleration)

Thanks to @rob-mcinerney for the idea!

**Comment by rob-mcinerney** [comment]
Test Photos!
Intellistep 0.0.50 - only mod is changing SYSCLK_FREQ in config_adv.h from 128 to 72mHz.

**Machine**
I am using a Anycubic Mega S with many mods, the biggest is the Hot End - changed to direct extrude instead of the Bowden, added a drag chain to the X axis to control the cables, all of this added some mass to the head which I believe contributed to the overshoot. I was hoping that the BTT S42B Servos (on X and Y) and their PID code could be used to tune the PID and therefore over come the additional mass - I like the direct extrude as I feel it does a better job of controlling the filament.

**Test**
Run an A/B test on the same part (xyz calibration cube - gcode available on request) as follows - (only changing the code in the X axis - Y axis still has the BTT code).
A. (Photo on LHS) - BTT stock Code at 100 mm/s - stopping the print at 80%, so the overshoot is easily identified, before the top layer covers it up.
B. (Photo on RHS) - Intellistep 0.0.50 with mod listed above, all other settings that same.

**Slicer Settings**
Used the same gcode file generated by PrusaSlicer (2.4.0), with these pertinent settings;
Acceleration at 1250 mm/s^2
Speed at 100mm/s on perimeters, external perimeters at 50%
Linear Advance - M900 K0.05
Junction Deviation - M205 J0.01

**Conclusion**
It is clear that the Intellistep code does a better job of keeping the X axis under control even w/o a PID loop - the photos show this if you look very carefully.

**Further testing** 
Install Intellistep code on the Y axis, then test to see if further tuning to Linear Advance and Junction Deviation is possible.
Try MKS Servo (will be here this afternoon).
Adjust the first layer settings.

![Top #2](https://user-images.githubusercontent.com/57737179/150652078-db00e3c8-7423-4137-93f5-e081c7ff906f.jpg)
![Side #2](https://user-images.githubusercontent.com/57737179/150652046-27c3ff45-719f-4e94-a8e5-cda9ae5c26d5.jpg)

Ignore the pry marks on the bottom of the part, just couldn't wait for the part to cool down!!!
Please pass on any thoughts!

**Reply by CAP1Sup** [reply]
Looking great! Did you see any layer misalignments (even small ones) over the print in the X-axis (when running Intellistep)? That’s the major issue remaining that’s halting the 0.1.0 release.

**Comment by rob-mcinerney** [comment]
This is a picture of the opposite of the Y side, if there was any misalignment they should show up here. I don’t see any. I see some ringing but that may be due to my infill. I will add this to the post tomorrow.

**Reply by CAP1Sup** [reply]
Glad to hear it! I’ll have to do some testing to see if removing the overclock fixes my issues as well. For your infill, did you see ringing like that before with your older Trinamic drivers? I’m curious as to if it’s a tuning issue or an Intellistep issue.

**Comment by rob-mcinerney** [comment]
Some ringing - but remember that is the Y side it is still the BTT Code.  The X motion side looks really good.  Photos tomorrow!

**Reply by rob-mcinerney** [reply]
Here are the photos.  First on is the X direction meaning that  it is the face opposite the X.
The second photo is the Y Direction.
These look really good! 
![X Direction](https://user-images.githubusercontent.com/57737179/150688704-cbc260dc-d37c-43d6-acc0-8197386e3b8a.jpg)
![Y Direction](https://user-images.githubusercontent.com/57737179/150688710-c7d75963-c8a3-46b9-a35b-4782ed11cc6e.jpg)

**Reply by CAP1Sup** [reply]
Glad to hear that Intellistep is working out for you! There doesn’t seem to be any layer shifting, which is great! I think that BTT’s code doesn’t ring like Intellistep because they’re not reproducing the steps exactly like Intellistep is. Klipper’s input shaping might be ideal for removing the issues you’re having with ringing. I need to do some testing, but hopefully reducing the clock should fix the miscounting for me as it did for you.
